web/src/engine/websites/NamiComi.ts

```diff
@@ -0,0 +1,111 @@
+import { Tags } from '../Tags';
+import icon from './NamiComi.webp';
+import { FetchJSON } from '../platform/FetchProvider';
+import { Page } from '../providers/MangaPlugin';
+import { DecoratableMangaScraper, type MangaPlugin, Manga, Chapter } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+
+type APIResult<T> = {
+    result: string,
+    data: T
+};
+
+type APIManga = {
+    id: string,
+    attributes: {
+        originalLanguage: string,
+        title: {
+            [lang: string]: string
+        }
+    }
+}
+
+type APIChapter = {
+    id: string,
+    attributes: {
+        volume: string,
+        chapter: string,
+        name: null | string,
+        translatedLanguage: string
+    }
+}
+
+type APIPages = {
+    baseUrl: string,
+    hash: string,
+    high: {
+        filename: string
+    }[]
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = 'https://api.namicomi.com';
+    private readonly mangaRegexp = new RegExp(`^${this.URI.origin}/en/title/(\[^/]+)\/[^/]+$`);
+    public constructor() {
+        super('namicomi', `NamiComi`, 'https://namicomi.com', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Comic, Tags.Language.Multilingual, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return this.mangaRegexp.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = url.match(this.mangaRegexp)[1];
+        const request = new Request(new URL(`/title/${id}`, this.apiUrl));
+        const { data } = await FetchJSON<APIResult<APIManga>>(request);
+        const title = (data.attributes.title[data.attributes.originalLanguage] || data.attributes.title.en).trim();
+        return new Manga(this, provider, data.id, title);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 0, run = true; run; page++) {
+            const mangas = await this.GetMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
```

Comment:
No need to get single pages for now. Website has `831` titles, so may just use a limit of `9999` to get all at once.

Reply:
ok